{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9bdf64a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import argparse\n",
    "from Bio import Entrez\n",
    "import pandas as pd\n",
    "from datetime import date\n",
    "from datetime import datetime\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "22a8372b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GEO_compare_test_2022_02_24_at_19_52'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "today = date.today()\n",
    "date = today.strftime(\"%Y_%m_%d\")\n",
    "now = datetime.now()\n",
    "time = now.strftime(\"at_%H_%M\")\n",
    "filename_prefix = \"GEO_compare_test_\" + str(date) + \"_\" + str(time)\n",
    "filename_prefix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "20970c9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "output = filename_prefix\n",
    "old = \"/Users/Cardiff/Downloads/HiChIP_Database_GEO/HiChIP_Databases.xlsx\"\n",
    "new = \"/Users/Cardiff/Downloads/HiChIP_Database_GEO/GEO_query_test_2022_02_24_at_16_33.xlsx\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "99651a0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "old_df = pd.read_excel(old)\n",
    "new_df = pd.read_excel(new)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4efa57d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def geo_compare(old, new, output):\n",
    "#     print(\"Arguements given are:\", old, new, output)\n",
    "#     old_df = pd.read_excel(old)\n",
    "#     new_df = pd.read_excel(new)\n",
    "#     # old_df[\"Year\"] = old_df[\"Year\"].astype(str)\n",
    "#     old_df=old_df.applymap(str)\n",
    "#     old_df[\"Year\"] = old_df[\"Year\"].str[:4]\n",
    "#     new_df=new_df.applymap(str)\n",
    "#     if \"Potential HiChIP\" not in old_df.columns:\n",
    "#         old_df[\"Potential HiChIP\"] = \"\"\n",
    "#         old_df = old_df[['Paper Title', 'Journal', 'Authors', 'Year', 'GEO / Data link',\n",
    "#            'Any other information', 'Organism', 'Tissue/Cell Line',\n",
    "#            'Potential HiChIP', 'Other matched data', 'Presenter']]\n",
    "#     merged_df = old_df.merge(new_df, how='outer', sort=False)\n",
    "#     merged_df.replace({'nan': \"\"}, inplace=True)\n",
    "#     print(merged_df)\n",
    "#     merged_df.to_excel(filename_prefix+\".xlsx\", index=False) \n",
    "#     print(\"Finished and saved to\", filename_prefix+\".xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4fd063cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# geo_compare(old, new, filename_prefix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "66cc475b",
   "metadata": {},
   "outputs": [],
   "source": [
    "GEO_pattern = re.compile('GSE[0123456789]+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "29d2970e",
   "metadata": {},
   "outputs": [],
   "source": [
    "old_GEOs = set()\n",
    "for x in old_df['GEO / Data link'].tolist():\n",
    "    GEO = GEO_pattern.findall(x)\n",
    "    old_GEOs.update(GEO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5cbef071",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_GEOs = set()\n",
    "for x in new_df['GEO / Data link'].tolist():\n",
    "    GEO = GEO_pattern.findall(x)\n",
    "    new_GEOs.update(GEO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0f303923",
   "metadata": {},
   "outputs": [],
   "source": [
    "differences_GEOs = new_GEOs.difference(old_GEOs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b354413f",
   "metadata": {},
   "outputs": [],
   "source": [
    "bools = []\n",
    "for x in new_df['GEO / Data link'].tolist():\n",
    "    if x in differences_GEOs: bools.append(True)\n",
    "    else: bools.append(False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c876f544",
   "metadata": {},
   "outputs": [],
   "source": [
    "differences_df = new_df.loc[bools]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "73025c0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/Cardiff/opt/anaconda3/lib/python3.9/site-packages/pandas/core/frame.py:5176: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().fillna(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Paper Title</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Authors</th>\n",
       "      <th>Year</th>\n",
       "      <th>GEO / Data link</th>\n",
       "      <th>Any other information</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Tissue/Cell Line</th>\n",
       "      <th>Potential HiChIP</th>\n",
       "      <th>Other matched data</th>\n",
       "      <th>Presenter</th>\n",
       "      <th>Date Added</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A primed immune transcriptional program is act...</td>\n",
       "      <td>10.1016/j.neuron.2021.12.034</td>\n",
       "      <td>Neuron</td>\n",
       "      <td>Meijer et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE166179</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Mus musculus; Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5065201: H3K27ac_HiChIP_IFN_mOPC_rep2\\nGSM5...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A small set of accessible enhancers enables br...</td>\n",
       "      <td>10.1093/nar/gkab1125</td>\n",
       "      <td>Nucleic Acids Res</td>\n",
       "      <td>Zaurin et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE179666</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5425946: aHiChIP PR T0\\nGSM5425945: aHiChIP...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CTCF HiChIP data in A673 Ewing sarcoma cell li...</td>\n",
       "      <td>10.1016/j.ccell.2021.04.001</td>\n",
       "      <td>Cancer Cell</td>\n",
       "      <td>Surdez et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE156650</td>\n",
       "      <td>Other</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM4735773: A673_siCT-Dh1_72h_HiChIP_CTCF_Rep1...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Dynamic chromatin regulatory landscape of huma...</td>\n",
       "      <td>10.1073/pnas.2104758118</td>\n",
       "      <td>Proc Natl Acad Sci U S A</td>\n",
       "      <td>Gennert et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE168881</td>\n",
       "      <td>Other</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Maybe</td>\n",
       "      <td>GSM5171871: Day10_CD19_H3K27ac_rep2\\nGSM517186...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Epigenomics landscape of colorectal cancer</td>\n",
       "      <td>10.1136/gutjnl-2020-322835</td>\n",
       "      <td>Gut</td>\n",
       "      <td>Orouji et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE136889</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM4053398: LS513 H3K27ac HiChIP\\nGSM4053399: ...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Expression profiling of Ewing sarcoma samples</td>\n",
       "      <td>10.1016/j.ccell.2021.04.001</td>\n",
       "      <td>Cancer Cell</td>\n",
       "      <td>Surdez et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE142162</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Maybe</td>\n",
       "      <td>GSM4221673: Ewing sarcoma sample EW150\\nGSM525...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Gibbin (AHDC1) regulates the mesoderm to patte...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2022</td>\n",
       "      <td>GSE180495</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5515078: HiChIP AHDC1/GIBBIN KO3\\nGSM551508...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>H3K27ac HiChIP of germinal center B cell-deriv...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE183797</td>\n",
       "      <td>Other</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5571213: H3K27ac_HiChIP_OCI-Ly7_control_rep...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Hi-ChIP dataset for epigenomics landscape of c...</td>\n",
       "      <td>10.1136/gutjnl-2020-322835</td>\n",
       "      <td>Gut</td>\n",
       "      <td>Orouji et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE136629</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM4053400: SW620 H3K27ac HiChIP\\nGSM4053397: ...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Locus-specific chromatin profiling of evolutio...</td>\n",
       "      <td>10.1093/nar/gkab1232</td>\n",
       "      <td>Nucleic Acids Res</td>\n",
       "      <td>Taylor et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE182304</td>\n",
       "      <td>Other</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td></td>\n",
       "      <td>Maybe</td>\n",
       "      <td>GSM5526450: 2102Ep_HiMeDIP\\nGSM5526451: E14_Hi...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Role of GR in protection of podocyte</td>\n",
       "      <td>10.1038/s42003-021-02209-8</td>\n",
       "      <td>Commun Biol</td>\n",
       "      <td>Wang et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE117888</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM3444790: H3K27ac-CTL1-HiChIP-seq\\nGSM344479...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>STAG2 promotes CTCF-anchored loop extrusion an...</td>\n",
       "      <td>10.1016/j.ccell.2021.04.001</td>\n",
       "      <td>Cancer Cell</td>\n",
       "      <td>Surdez et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE133228</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM4735778: A673_siSA2-Dh6_72h_HiChIP_CTCF_Rep...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Single-nucleotide-level mapping of DNA regulat...</td>\n",
       "      <td>10.1038/s41588-021-00861-8</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Cheng et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE168719</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5159894: Hudep2_D3_H3K27AC_HiCHIP</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Specific genomic features underlie the co-opti...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2022</td>\n",
       "      <td>GSE192951</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5769506: HICHIP_no_DOX_rep3\\nGSM5769503: HI...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Systematic Functional Characterization of Anti...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE148958</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Mus musculus; Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5533484: HiChIP_SPT6_LNA_V5\\nGSM5533482: Hi...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>The Xia-Gibbs Syndrome protein Gibbin regulate...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2022</td>\n",
       "      <td>GSE181971</td>\n",
       "      <td>Other</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5515080: HiChIP WT2\\nGSM5515077: HiChIP AHD...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>UTX condensation underlies its tumor suppressi...</td>\n",
       "      <td>10.1038/s41586-021-03903-7</td>\n",
       "      <td>Nature</td>\n",
       "      <td>Shi et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE149420</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>GSM5379698: HiChIP_H3K27ac_WT\\nGSM5379695: HiC...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>cLoops2: a full-stack comprehensive analytical...</td>\n",
       "      <td>10.1093/nar/gkab1233</td>\n",
       "      <td>Nucleic Acids Res</td>\n",
       "      <td>Cao et al.</td>\n",
       "      <td>2021</td>\n",
       "      <td>GSE179010</td>\n",
       "      <td>Other</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td></td>\n",
       "      <td>Maybe</td>\n",
       "      <td>GSM5403564: ChIC_GM12878_IgG_Rep1\\nGSM5403558:...</td>\n",
       "      <td></td>\n",
       "      <td>2022_02_24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Paper Title  \\\n",
       "0   A primed immune transcriptional program is act...   \n",
       "1   A small set of accessible enhancers enables br...   \n",
       "4   CTCF HiChIP data in A673 Ewing sarcoma cell li...   \n",
       "7   Dynamic chromatin regulatory landscape of huma...   \n",
       "9          Epigenomics landscape of colorectal cancer   \n",
       "11      Expression profiling of Ewing sarcoma samples   \n",
       "13  Gibbin (AHDC1) regulates the mesoderm to patte...   \n",
       "14  H3K27ac HiChIP of germinal center B cell-deriv...   \n",
       "15  Hi-ChIP dataset for epigenomics landscape of c...   \n",
       "22  Locus-specific chromatin profiling of evolutio...   \n",
       "24               Role of GR in protection of podocyte   \n",
       "25  STAG2 promotes CTCF-anchored loop extrusion an...   \n",
       "26  Single-nucleotide-level mapping of DNA regulat...   \n",
       "27  Specific genomic features underlie the co-opti...   \n",
       "28  Systematic Functional Characterization of Anti...   \n",
       "30  The Xia-Gibbs Syndrome protein Gibbin regulate...   \n",
       "34  UTX condensation underlies its tumor suppressi...   \n",
       "35  cLoops2: a full-stack comprehensive analytical...   \n",
       "\n",
       "                             DOI                   Journal         Authors  \\\n",
       "0   10.1016/j.neuron.2021.12.034                    Neuron   Meijer et al.   \n",
       "1           10.1093/nar/gkab1125         Nucleic Acids Res   Zaurin et al.   \n",
       "4    10.1016/j.ccell.2021.04.001               Cancer Cell   Surdez et al.   \n",
       "7        10.1073/pnas.2104758118  Proc Natl Acad Sci U S A  Gennert et al.   \n",
       "9     10.1136/gutjnl-2020-322835                       Gut   Orouji et al.   \n",
       "11   10.1016/j.ccell.2021.04.001               Cancer Cell   Surdez et al.   \n",
       "13                                                                           \n",
       "14                                                                           \n",
       "15    10.1136/gutjnl-2020-322835                       Gut   Orouji et al.   \n",
       "22          10.1093/nar/gkab1232         Nucleic Acids Res   Taylor et al.   \n",
       "24    10.1038/s42003-021-02209-8               Commun Biol     Wang et al.   \n",
       "25   10.1016/j.ccell.2021.04.001               Cancer Cell   Surdez et al.   \n",
       "26    10.1038/s41588-021-00861-8                 Nat Genet    Cheng et al.   \n",
       "27                                                                           \n",
       "28                                                                           \n",
       "30                                                                           \n",
       "34    10.1038/s41586-021-03903-7                    Nature      Shi et al.   \n",
       "35          10.1093/nar/gkab1233         Nucleic Acids Res      Cao et al.   \n",
       "\n",
       "    Year GEO / Data link                              Any other information  \\\n",
       "0   2021       GSE166179  Genome binding/occupancy profiling by high thr...   \n",
       "1   2021       GSE179666  Expression profiling by high throughput sequen...   \n",
       "4   2021       GSE156650                                              Other   \n",
       "7   2021       GSE168881                                              Other   \n",
       "9   2021       GSE136889  Genome binding/occupancy profiling by high thr...   \n",
       "11  2021       GSE142162                      Expression profiling by array   \n",
       "13  2022       GSE180495  Expression profiling by high throughput sequen...   \n",
       "14  2021       GSE183797                                              Other   \n",
       "15  2021       GSE136629  Genome binding/occupancy profiling by high thr...   \n",
       "22  2021       GSE182304                                              Other   \n",
       "24  2021       GSE117888  Expression profiling by high throughput sequen...   \n",
       "25  2021       GSE133228  Expression profiling by high throughput sequen...   \n",
       "26  2021       GSE168719  Genome binding/occupancy profiling by high thr...   \n",
       "27  2022       GSE192951  Expression profiling by high throughput sequen...   \n",
       "28  2021       GSE148958  Expression profiling by high throughput sequen...   \n",
       "30  2022       GSE181971                                              Other   \n",
       "34  2021       GSE149420  Genome binding/occupancy profiling by high thr...   \n",
       "35  2021       GSE179010                                              Other   \n",
       "\n",
       "                      Organism Tissue/Cell Line Potential HiChIP  \\\n",
       "0   Mus musculus; Homo sapiens                               Yes   \n",
       "1                 Homo sapiens                               Yes   \n",
       "4                 Homo sapiens                               Yes   \n",
       "7                 Homo sapiens                             Maybe   \n",
       "9                 Homo sapiens                               Yes   \n",
       "11                Homo sapiens                             Maybe   \n",
       "13                Homo sapiens                               Yes   \n",
       "14                Homo sapiens                               Yes   \n",
       "15                Homo sapiens                               Yes   \n",
       "22  Homo sapiens; Mus musculus                             Maybe   \n",
       "24                Homo sapiens                               Yes   \n",
       "25                Homo sapiens                               Yes   \n",
       "26                Homo sapiens                               Yes   \n",
       "27                Homo sapiens                               Yes   \n",
       "28  Mus musculus; Homo sapiens                               Yes   \n",
       "30                Homo sapiens                               Yes   \n",
       "34  Homo sapiens; Mus musculus                               Yes   \n",
       "35                Homo sapiens                             Maybe   \n",
       "\n",
       "                                   Other matched data Presenter  Date Added  \n",
       "0   GSM5065201: H3K27ac_HiChIP_IFN_mOPC_rep2\\nGSM5...            2022_02_24  \n",
       "1   GSM5425946: aHiChIP PR T0\\nGSM5425945: aHiChIP...            2022_02_24  \n",
       "4   GSM4735773: A673_siCT-Dh1_72h_HiChIP_CTCF_Rep1...            2022_02_24  \n",
       "7   GSM5171871: Day10_CD19_H3K27ac_rep2\\nGSM517186...            2022_02_24  \n",
       "9   GSM4053398: LS513 H3K27ac HiChIP\\nGSM4053399: ...            2022_02_24  \n",
       "11  GSM4221673: Ewing sarcoma sample EW150\\nGSM525...            2022_02_24  \n",
       "13  GSM5515078: HiChIP AHDC1/GIBBIN KO3\\nGSM551508...            2022_02_24  \n",
       "14  GSM5571213: H3K27ac_HiChIP_OCI-Ly7_control_rep...            2022_02_24  \n",
       "15  GSM4053400: SW620 H3K27ac HiChIP\\nGSM4053397: ...            2022_02_24  \n",
       "22  GSM5526450: 2102Ep_HiMeDIP\\nGSM5526451: E14_Hi...            2022_02_24  \n",
       "24  GSM3444790: H3K27ac-CTL1-HiChIP-seq\\nGSM344479...            2022_02_24  \n",
       "25  GSM4735778: A673_siSA2-Dh6_72h_HiChIP_CTCF_Rep...            2022_02_24  \n",
       "26               GSM5159894: Hudep2_D3_H3K27AC_HiCHIP            2022_02_24  \n",
       "27  GSM5769506: HICHIP_no_DOX_rep3\\nGSM5769503: HI...            2022_02_24  \n",
       "28  GSM5533484: HiChIP_SPT6_LNA_V5\\nGSM5533482: Hi...            2022_02_24  \n",
       "30  GSM5515080: HiChIP WT2\\nGSM5515077: HiChIP AHD...            2022_02_24  \n",
       "34  GSM5379698: HiChIP_H3K27ac_WT\\nGSM5379695: HiC...            2022_02_24  \n",
       "35  GSM5403564: ChIC_GM12878_IgG_Rep1\\nGSM5403558:...            2022_02_24  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "differences_df.fillna(\"\",inplace=True)\n",
    "differences_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d591d27e",
   "metadata": {},
   "outputs": [],
   "source": [
    "differences_df.to_excel(output+\".xlsx\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
